js/apps/system/_admin/aardvark/APP/frontend/js/views/graphViewer.js

```diff
@@ -1063,12 +1063,22 @@
 
       var callback = function (error, id, msg) {
         if (!error) {
-          var edge = {
-            source: from,
-            target: to,
-            id: id,
-            color: self.graphConfig.edgeColor || self.ecolor
-          };
+          var edge;
+          try {
+            edge = this.editor.get();
+          } catch (x) {
+            arangoHelper.arangoError("failed to parse JSON document", x.message);
+            return;
+          }
+          try {
+            edge.source = from;
+          } catch (x) {
+            edge = {};
+            edge.source = from;
+          }
```

Comment:
this looks a bit strange... are we trying to enforce edge is an object?

Reply:
yes.